solutions/src/main/java/ru/sberbank/school/task08/KryoManager.java

```diff
@@ -0,0 +1,80 @@
+package ru.sberbank.school.task08;
+
+import com.esotericsoftware.kryo.KryoException;
+import lombok.NonNull;
+import ru.sberbank.school.task08.state.GameObject;
+import ru.sberbank.school.task08.state.InstantiatableEntity;
+import ru.sberbank.school.task08.state.MapState;
+import ru.sberbank.school.task08.state.Savable;
+import ru.sberbank.school.util.Solution;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.Serializer;
+import com.esotericsoftware.kryo.io.Input;
+import com.esotericsoftware.kryo.io.Output;
+
+@Solution(8)
+public class KryoManager extends SaveGameManager<MapState<GameObject>, GameObject> {
+    private Kryo kryo;
+
+    public KryoManager(@NonNull String filesDirectoryPath) {
+        super(filesDirectoryPath);
+    }
+
+    @Override
+    public void initialize() {
+        kryo = new Kryo();
+        KryoSerializer kryoSerializer = new KryoSerializer();
+        kryo.register(MapState.class, kryoSerializer);
+        kryo.register(InstantiatableEntity.Type.class);
+        kryo.register(InstantiatableEntity.Status.class);
+        kryo.register(GameObject.class);
+        kryo.register(ArrayList.class);
+    }
+
+    @Override
+    public void saveGame(@NonNull String filename, @NonNull MapState<GameObject> gameState) throws SaveGameException {
+        try (FileOutputStream fos = new FileOutputStream(filesDirectory + File.separator + filename);
+             Output out = new Output(fos)) {
+            kryo.writeObject(out, gameState);
+        } catch (NullPointerException | FileNotFoundException ex) {
+            throw new SaveGameException("File not found");
+        } catch (KryoException e) {
+            throw new SaveGameException("Kryo ex");
+        } catch (IOException ex) {
+            ex.printStackTrace();
+        }
+    }
+
+    @Override
+    public MapState<GameObject> loadGame(@NonNull String filename) throws SaveGameException {
+        try (FileInputStream fis = new FileInputStream(filesDirectory + File.separator + filename);
+             Input in = new Input(fis)) {
```

Comment:
&#39;try resource&#39; has incorrect indentation level 13, expected level should be 16.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-1986486537>*